flink-ml-parent/flink-ml-lib/pom.xml

```diff
@@ -57,4 +60,30 @@ under the License.
 			<version>1.1.2</version>
 		</dependency>
 	</dependencies>
+
+	<build>
+		<plugins>
+			<!-- Build flink-ml-lib jar -->
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-shade-plugin</artifactId>
```

Comment:
I would move this (and the corresponding NOTICE) into the `uber` module.

The basic idea is that the `uber` module is responsible for all things related to the distribution. The individual modules usually shouldn't have to shade dependencies; users should set flink-ml dependencies to provided anyways so we don't have to concerned about them accidently bundling this dependency.